src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -666,21 +676,34 @@ public static void joinGame(final GameDescription description, final Messengers
         return;
       }
     }
-    joinGame(description.getPort(), description.getHostedBy().getAddress().getHostAddress(), newClassPath, messengers);
+    joinGame(description.getPort(), description.getHostedBy().getAddress().getHostAddress(),
+        areSameVersion ? "" : newClassPath, messengers);
   }
 
   // newClassPath can be null
   private static void joinGame(final int port, final String hostAddressIP, final String newClassPath,
       final Messengers messengers) {
     final List<String> commands = new ArrayList<>();
-    ProcessRunnerUtil.populateBasicJavaArgs(commands, newClassPath);
-    commands.add("-D" + TRIPLEA_CLIENT_PROPERTY + "=true");
-    commands.add("-D" + TRIPLEA_PORT_PROPERTY + "=" + port);
-    commands.add("-D" + TRIPLEA_HOST_PROPERTY + "=" + hostAddressIP);
-    commands.add("-D" + TRIPLEA_NAME_PROPERTY + "=" + messengers.getMessenger().getLocalNode().getName());
-    final String javaClass = "games.strategy.engine.framework.GameRunner";
-    commands.add(javaClass);
-    ProcessRunnerUtil.exec(commands);
+    final boolean startInNewProcess = !("".equals(newClassPath) && MainFrame.getInstance() != null);
+    String prefix = startInNewProcess ? "-D" : "";
+    if (startInNewProcess) {
+      ProcessRunnerUtil.populateBasicJavaArgs(commands, newClassPath);
```

Comment:
Note: ProcessRunnerUtil has a lot of problems. It seems like it can be entirely inlined to this class. In the future IMO that would be a really good refactor followed by re-breaking up the class into functional components.